src/field/gf_101_2.rs

```diff
@@ -0,0 +1,335 @@
+use core::{
+  iter::{Product, Sum},
+  ops::{Add, AddAssign, Mul, Neg, Sub, SubAssign},
+};
+use std::ops::{Div, DivAssign, MulAssign, Rem};
+
+use serde::{Deserialize, Serialize};
+
+use super::ExtensionField;
+use crate::field::FiniteField;
+
+/// Pluto curve with modulus 101 supports two degree extension field. This can be verified
+/// by finding out embedding degree of the curve, i.e. smallest k: r|q^k-1
+const EXT_DEGREE: usize = 2;
+const QUADRATIC_EXTENSION_FIELD_ORDER: u32 = 101 * 101;
+
+#[derive(Clone, Default, Copy, Eq, PartialEq, Hash, Debug, Serialize, Deserialize)]
+
+/// Quadratic Extension field element represented as polynomial of degree 1 in form:
+/// a_0 + a_1*t where {a_0, a_1} \in \mathhbb{F}. Uses irreducible poly of the form:
```

Comment:
I think this is the right approach but curious how or if we should use this in our polynomial module. I know @Autoparallel you have been working on the FFTs and polynomial interface. Maybe you have some insight here?

Reply:
Let me get the polynomial PR up.